drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -118,46 +119,46 @@ static int pwm_nrf5_sw_pin_set(struct device *dev, u32_t pwm,
 	ppi_index = config->ppi_base + (channel << 1);
 	NRF_PPI->CHENCLR = BIT(ppi_index) | BIT(ppi_index + 1);
 
-	/* configure GPIO pin as output */
-	NRF_GPIO->DIRSET = BIT(pwm);
 	if (pulse_cycles == 0) {
 		/* 0% duty cycle, keep pin low */
-		NRF_GPIO->OUTCLR = BIT(pwm);
-
+		NRF_GPIOTE->CONFIG[config->gpiote_base + channel] = 0x00000003 |
```

Comment:
Is it really necessary to use GPIOTE in task mode (and upto 0.5 uA current) for 0% and 100% duty cycle? Revert back, so that a normal GPIO can do the job with no added current consumption.

Reply:
I think this is the cleanest solution (and it solves the bug in the current implementation when going from 0 < duty cycle < 100 to either 0% or 100% duty cycle, as noted in the commit message).

Reverting to using normal GPIO will require other changes (e.g. disabling the GPIOTE when either 0% or 100% duty cycle is requested, otherwise the normal GPIO settings will have no effect).

If you feel strongly about saving the 0.5 uA, I will work on a GPIO-based solution.